build.gradle

```diff
@@ -201,44 +202,46 @@ project(':iceberg-bundled-guava') {
 
 project(':iceberg-api') {
   dependencies {
-    compile project(path: ':iceberg-bundled-guava', configuration: 'shadow')
-    compileOnly "com.google.errorprone:error_prone_annotations:2.3.3"
-    testCompile "org.apache.avro:avro"
+    implementation project(path: ':iceberg-bundled-guava', configuration: 'shadow')
+    compileOnly "com.google.errorprone:error_prone_annotations"
+    testImplementation "org.apache.avro:avro"
   }
 }
 
 project(':iceberg-common') {
   dependencies {
-    compile project(path: ':iceberg-bundled-guava', configuration: 'shadow')
+    implementation project(path: ':iceberg-bundled-guava', configuration: 'shadow')
   }
 }
 
 project(':iceberg-core') {
   dependencies {
-    compile project(':iceberg-api')
-    compile project(':iceberg-common')
+    implementation project(':iceberg-api')
+    implementation project(':iceberg-common')
+    implementation project(path: ':iceberg-bundled-guava', configuration: 'shadow')
 
-    compile("org.apache.avro:avro") {
+    implementation("org.apache.avro:avro") {
       exclude group: 'org.tukaani' // xz compression is not supported
     }
 
-    compile "com.fasterxml.jackson.core:jackson-databind"
-    compile "com.fasterxml.jackson.core:jackson-core"
-    compile "com.github.ben-manes.caffeine:caffeine"
+    implementation "com.fasterxml.jackson.core:jackson-databind"
+    implementation "com.fasterxml.jackson.core:jackson-core"
+    implementation "com.github.ben-manes.caffeine:caffeine"
     compileOnly("org.apache.hadoop:hadoop-client") {
       exclude group: 'org.apache.avro', module: 'avro'
       exclude group: 'org.slf4j', module: 'slf4j-log4j12'
     }
 
-    testCompile "org.xerial:sqlite-jdbc"
-    testCompile project(path: ':iceberg-api', configuration: 'testArtifacts')
+    testImplementation "org.xerial:sqlite-jdbc"
+    testImplementation project(path: ':iceberg-api', configuration: 'testArtifacts')
   }
 }
 
 project(':iceberg-data') {
   dependencies {
-    compile project(':iceberg-api')
-    compile project(':iceberg-core')
+    implementation project(path: ':iceberg-bundled-guava', configuration: 'shadow')
+    implementation project(':iceberg-api')
```

Comment:
I think this should be an `api` dependency because these classes are used in the API of `iceberg-data`. For example, `iceberg-api` includes the expressions, which are passed into `IcebergGenerics`.

Reply:
sgtm